bot/cogs/moderation/infractions.py

```diff
@@ -235,8 +235,32 @@ async def apply_ban(self, ctx: Context, user: UserSnowflake, reason: str, **kwar
 
         Will also remove the banned user from the Big Brother watch list if applicable.
         """
-        if await utils.has_active_infraction(ctx, user, "ban"):
-            return
+        # In the case of a permanent ban, we don't need get_active_infractions to tell us if one is active
+        send_msg = "expires_at" in kwargs
```

Comment:
Since you explicitly check for `None` elsewhere, should you be doing that here too?

Reply:
Addressed in most recent commit